examples/speaker_recognition/voxceleb_eval.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import sys
+
+import numpy as np
+from scipy.interpolate import interp1d
+from scipy.optimize import brentq
+from sklearn.metrics import roc_curve
+from tqdm import tqdm
+
+
+"""
+This script faciliates to get EER % based on cosine-smilarity 
+for Voxceleb dataset.
+
+Args:
+    trial_file str: path to voxceleb trial file
+    emb : path to numpy file of embeddings (generated from spkr_get_emb.py)
+    manifest: path to test manifest file that contains path of audio files which is used in spkr_get_emb.py
+    Note: order of audio files in manifest file should match the embeddings
+"""
+
+
+def get_labels(manifest):
+    lines = open(manifest, 'r').readlines()
+    test_list = {}
+    for idx, line in enumerate(lines):
+        line = line.strip()
+        dic = json.loads(line)
+        structure = dic['audio_filepath'].split('/')[-3:]
+        uniq_name = '@'.join(structure)
+        if uniq_name in test_list:
+            raise KeyError("uniq name is already present")
+        test_list[uniq_name] = idx
+    return test_list
+
+
+def get_acc(trial_file='', emb='', manifest=''):
+
+    X_test = np.load(emb)
+    manifest_lines = open(manifest, 'r').readlines()
+    assert len(X_test) == len(manifest_lines)
+
+    test_list = get_labels(manifest)
+
+    tmp_file = open(trial_file, 'r').readlines()
+    trail_score = open('trial_score.txt', 'w')
+
+    trial_embs = []
+    keys = []
+    all_scores = []
+    all_keys = []
+
+    # for each of trails in trial file
+    for line in tqdm(tmp_file):
+        line = line.strip()
+        truth, x_speaker, y_speaker = line.split()
+
+        x_speaker = x_speaker.split('/')
+        x_speaker = '@'.join(x_speaker)
+
+        y_speaker = y_speaker.split('/')
+        y_speaker = '@'.join(y_speaker)
+
+        x_idx = test_list[x_speaker]
+        y_idx = test_list[y_speaker]
+
+        X = X_test[x_idx]
+        Y = X_test[y_idx]
+
+        if x_speaker not in keys:
+            keys.append(x_speaker)
+            trial_embs.extend([X])
+
+        if y_speaker not in keys:
+            keys.append(y_speaker)
+            trial_embs.extend([Y])
+
```

Comment:
Looks like keys and trial_embs are only used for KALDI PLDA. can you pass an argument like save_kaldi_emb and put this part of the code under `if save_kaldi_emb:` . You could also uncomment the code below and put it under the same condition. 

May be also store keys and trial_embs in a dict, instead of traversing the keys list every line?

Reply:
Thanks for this suggestion, was planning to move embs to dict and key based, done now. Added save_kaldi_emb flag too

Reply:
I think you could also use trial_emb_dict and use it like this:
`if save_kaldi_emb and x_speaker not in trial_emb_dict:
                trial_emb_dict[x_speaker] = X`
instead of list lookup `x_speaker not in keys` .